src/lib/Hydra/Controller/Root.pm

```diff
@@ -529,5 +529,20 @@ sub log :Local :Args(1) {
         notFound($c, "The build log of $drvPath is not available.");
     }
 }
+sub runcommandlog :Local :Args(1) {
+    my ($self, $c, $uuid) = @_;
+
+    my $tail = $c->request->params->{"tail"};
+
+    die if defined $tail && $tail !~ /^[0-9]+$/;
+
+    my $logFile = constructRunCommandLogPath($uuid);
```

Comment:
I think we should probably pass the `RunCommandLog` Result class to `constructRunCommandLogPath`, and let constructRunCommandLogPath pick the UUID out. This will make sure that the record must exist for us to try and calculate a file for it. This helps avoid any spookiness about filesystem traversal and user input (which is still a problem with this bit of code.)

Reply:
Done in 591ae4a0fa02a35ad354ab615cedd1458761af8e -- could you confirm that my implementation does indeed avoid said spookiness?